gtsfm/data_association/data_assoc.py

```diff
@@ -128,6 +128,14 @@ def run(
 
         # dump the 3d point cloud before Bundle Adjustment for offline visualization
         points_3d = [list(triangulated_data.get_track(j).point3()) for j in range(num_accepted_tracks)]
+
+        #create the centered and rotated 3d point cloud - properly aligned with x,y,z axes
+        centered_pc = ellipsoid_utils.center_point_cloud(points)
+        hullSet_points = ellipsoid_utils.compute_convex_hull(centered_pc)
+        polynomial_params = ellipsoid_utils.fit_ls_ellipsoid(hullSet_points)
+        center, axes, rot_matrix = ellipsoid_utils.extract_params_from_poly(polynomial_params)
+        aligned_points_3d = ellipsoid_utils.apply_ellipsoid_rotation(rot_matrix, centered_pc)
```

Comment:
I am surprised we don’t have utils to translate and rotate a point cloud already. If we really don’t, we should expand the scope of “ellipsoid_utils” to something like “3d_geometry_utils” and provide those utils. Also, we should use gtsam types for Rotation and translation, along with naming conventions like “wRc” that are used in other parts of GTSfM.